subprojects/test-runner/instrumentation-tests/src/main/kotlin/com/avito/instrumentation/internal/report/listener/ReportProcessorImpl.kt

```diff
@@ -52,85 +52,79 @@ internal class ReportProcessorImpl(
                             because = "There is not enough context here about cause",
                             possibleSolutions = listOf(
                                 "MBS-11279 should help with correct error message from AdbDevice",
                                 "MBS-11281 to return tests with such errors back to retry queue"
                             ),
-                            cause = throwable
+                            throwable = throwable
                         )
 
                         processFailure(
-                            throwable = problem,
+                            problem = problem,
                             testStaticData = testFromSuite,
                             logcatBuffer = logcatBuffer
                         )
                     }.getOrThrow()
 
             is TestResult.Incomplete ->
                 with(result.infraError) {
-                    logger.warn("${error.message} while executing ${test.testName}", this.error)
+
+                    val problemBuilder = Problem.Builder(
+                        shortDescription = "Can't complete test execution",
+                        context = "ReportProcessor handling incomplete test result",
+                    )
+
+                    problemBuilder.throwable(error)
 
                     when (this) {
-                        is TestCaseRun.Result.Failed.InfrastructureError.FailedOnParsing ->
+                        is TestCaseRun.Result.Failed.InfrastructureError.FailedOnParsing -> {
                             metricsSender.sendFailedOnParsingInstrumentation()
+                            problemBuilder.because("Can't parse instrumentation output, see underlying exception")
+                        }
 
-                        is TestCaseRun.Result.Failed.InfrastructureError.FailedOnStart ->
+                        is TestCaseRun.Result.Failed.InfrastructureError.FailedOnStart -> {
                             metricsSender.sendFailedOnStartInstrumentation()
+                            problemBuilder.because("Can't start test")
+                        }
 
-                        is TestCaseRun.Result.Failed.InfrastructureError.Timeout ->
+                        is TestCaseRun.Result.Failed.InfrastructureError.Timeout -> {
                             metricsSender.sendTimeOut()
+                            problemBuilder.because(
+                                "Test didn't finish in time. " +
+                                    "Test Runner has hardcoded timeout of $timeoutMin minutes"
+                            )
+                        }
 
-                        is TestCaseRun.Result.Failed.InfrastructureError.Unexpected ->
+                        is TestCaseRun.Result.Failed.InfrastructureError.Unexpected -> {
                             metricsSender.sendUnexpectedInfraError()
+                        }
                     }
 
+                    val problem = problemBuilder.build()
```

Comment:
Do we send this failure to the ReportViewer?

Reply:
Only message? What about a stack trace?

Reply:
Stacktrace passed as before, nothing changed in this regard